tests/system/marathon_common_tests.py

```diff
@@ -1241,3 +1243,174 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
+    """Tests if the applications with a `region` and a `zone` defined are always starting on the
+       same region and zone
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region ({})".format(
+        region,
+    )
+
+    single_zone = zones[0]
+
+    # Create an app pinned on the given region and zone.
+    #
+    # (We are starting more instances than the number of zones in the region in order to increase
+    #  the chance of a faulty instance to appear on a different zone)
+    #
+    app_def = apps.faultdomain_app(
+        region=region,
+        zone=single_zone,
+        instances=len(zones) * 2,
+        suffix="-region-and-zone")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=(single_zone,))
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_region_unique():
+    """Tests if the applications with a ["@region", "UNIQUE"] constraint correctly starts the apps
+       across all regions in the cluster
+    """
+
+    # Find out all the regions in the cluster
+    regions = set(get_all_cluster_regions().keys())
+
+    assert len(regions) > 1, "The tests require at least 2 regions in the cluster"
+
+    # Create an app that should be deployed on a unique region
+    # The number of instances should be the number of regions available in the system
+    app_def = apps.faultdomain_app(
+        instances=len(regions),
+        constraints=[["@region", "UNIQUE"]],
+        suffix="-region-unique"
+    )
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(app_id=app_def['id'])
+
+    # Check if the application was launched correctly
+    app = client.get_app(app_def['id'])
+    (used_regions, used_zones) = get_used_regions_and_zones(get_app_domains(app))
+
+    assert used_regions == set(regions), "App {} was not launched in all regions: {} instead of {}".format(
+            app['id'], ', '.join(used_regions), ', '.join(regions)
+        )
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_zone_unique():
+    """Tests if the applications with a region fixed and a ["@zone", "UNIQUE"] constraint correctly starts
+       the apps across all zones in the given region
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
```

Comment:
Why do we have to find the biggest region?

Reply:
Because some tests require at least 2 zones in a region. For these tests, this function makes sure we pick the correct one.